### ISSUE TITLE
Satellizer executes multiple requests to the same application endpoints

### ISSUE THREAD

**Issue by akkie** [sahat/satellizer#287]
I've a problem to authenticate to Xing via OAuth1. The problem is that satellizer sends the same requests twice to the backend. 

As example for the Xing request:
`GET /authenticate/xing?oauth_token=f647627c85910bß2h8k3&oauth_verifier=6163`

This is the request which will be send to the backend and which initializes the retrieval of the access token. The problem is now that the request will be sent twice to the backend which then also causes the backend to retrieve the access token twice. I think that Xing now invalidates the request token after the first access token was received. So on the second request to the access token endpoint, Xing responds with an error that the request token is invalid.

Twitter doesn't have the problem of the invalid request token. But satellizer sends also the request twice.

Any suggestions?

**Comment by akkie** [comment]
After digging a bit deeper into satellizer I think I've found the issue. In the popup the complete OAuth flow is handled with all redirects. So the last redirect is to URL:

`GET /authenticate/xing?oauth_token=f647627c85910bß2h8k3&oauth_verifier=6163`

which returns also the JWT satellizer depends on.

But now after the popup was closed satellizer requests the URL again to retrieve the JWT. Is it not possible to grab the token from the popup instead of requesting the token again?

**Comment by kfuchs** [comment]
It seems now twitter does have an issue when you request the token again. Twitter no longer works, was this ever fixed? I'm still using ver 10.1.

**Comment by akkie** [comment]
Same problem here with the Twitter support. For the old version(0.8.8) which I have currently implemented I run now into this issue. With the newest version (0.11.13) I run into the [open popup issue](https://github.com/sahat/satellizer/issues/417).

So Twitter support is no completely broken for me.

**Comment by kfuchs** [comment]
@akkie 
My current work around is to simply cache it on the back-end for like a minute. This way the 2nd api call is just returned the results of the first. Not perfect but does the trick for now.

**Comment by akkie** [comment]
@kfuchs Yes that fixed it for me too. Anyway, it would be nice if this issue gets fixed.

**Comment by CrackDavid** [comment]
@akkie Is there a possibility you may share your Code with us? I am currently developing a Xing-API Call-Structure and that would help a lot :)

**Comment by akkie** [comment]
@CrackDavid Sure, but I'm not sure if it helps because it's written in Scala.

https://github.com/mohiva/play-silhouette-angular-seed/blob/master/app/controllers/SocialAuthController.scala#L81
